### PR TITLE
[SPARK-41332][CONNECT][PYTHON]  Fix `nullOrdering` in `SortOrder`

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
fix the wrong `nullOrdering`

### Why are the changes needed?
existing implementation generate incorrect `nullOrdering`;

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
added ut

### CODE REVIEW

File: python/pyspark/sql/tests/connect/test_connect_basic.py [hunk 1]
```diff
@@ -411,6 +411,48 @@ def test_subquery_alias(self) -> None:
         )
         self.assertTrue("special_alias" in plan_text)
 
+    def test_sort(self):
```
Comment (zhengruifeng) [inline python/pyspark/sql/tests/connect/test_connect_basic.py:414]: these tests fails before this PR, we should add more e2e tests.

File: python/pyspark/sql/connect/plan.py [hunk 2]
```diff
@@ -492,40 +492,27 @@ class Sort(LogicalPlan):
...
-                proto.Sort.SortNulls.SORT_NULLS_FIRST
-                if not col._expr.nullsLast
-                else proto.Sort.SortNulls.SORT_NULLS_LAST
-            )
-            return sf
-        else:
-            sf = proto.Sort.SortField()
-            # Check string
-            if isinstance(col, Column):
-                sf.expression.CopyFrom(col.to_plan(session))
```
Comment (zhengruifeng) [inline python/pyspark/sql/connect/plan.py:523]: this path is not reachable

File: python/pyspark/sql/connect/plan.py [hunk 3]
```diff
@@ -492,40 +492,27 @@ class Sort(LogicalPlan):
...
+            if isinstance(col._expr, SortOrder):
+                sort = col._expr
             else:
-                sf.expression.CopyFrom(self.unresolved_attr(col))
-            sf.direction = proto.Sort.SortDirection.SORT_DIRECTION_ASCENDING
-            sf.nulls = proto.Sort.SortNulls.SORT_NULLS_LAST
```
Comment (zhengruifeng) [inline python/pyspark/sql/connect/plan.py:527]: wrong default null ordering

File: python/pyspark/sql/connect/column.py [hunk 4]
```diff
@@ -307,7 +305,21 @@ def __str__(self) -> str:
         return str(self.ref) + " ASC" if self.ascending else " DESC"
 
     def to_plan(self, session: "SparkConnectClient") -> proto.Expression:
-        return self.ref.to_plan(session)
+        # TODO: move SortField from relations.proto to expression.proto
```
Comment (amaliujia) [inline python/pyspark/sql/connect/column.py:308]: Nit: better to have a JIRA link here. I usually do TODO(SPARK-xxxx)

File: python/pyspark/sql/tests/connect/test_connect_basic.py (hunk 1 above)
Comment (amaliujia) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:414]: big +1

File: python/pyspark/sql/tests/connect/test_connect_basic.py [hunk 5]
```diff
@@ -411,6 +411,48 @@ def test_subquery_alias(self) -> None:
...
+            cdf.sort(cdf.c, "b").toPandas(),
+            sdf.sort(sdf.c, "b").toPandas(),
+        )
+        self.assert_eq(
+            cdf.sort(cdf.c.desc(), "b").toPandas(),
+            sdf.sort(sdf.c.desc(), "b").toPandas(),
+        )
+        self.assert_eq(
+            cdf.sort(cdf.c.desc(), cdf.a.asc()).toPandas(),
+            sdf.sort(sdf.c.desc(), sdf.a.asc()).toPandas(),
```
Comment (amaliujia) [inline python/pyspark/sql/tests/connect/test_connect_basic.py:453]: we don't have null last and null first test?

File: python/pyspark/sql/connect/column.py (hunk 4 above)
Comment (zhengruifeng) [reply python/pyspark/sql/connect/column.py:308]: nice, will update

File: python/pyspark/sql/tests/connect/test_connect_basic.py (hunk 5 above)
Comment (zhengruifeng) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:453]: no, never test it before.

let me also add some tests in plan-only test

**Review by HyukjinKwon — APPROVED** [review]

**Review by amaliujia — APPROVED** [review]
LGTM once test pass

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**